src/main/java/ch/njol/skript/events/EvtAdvancement.java

```diff
@@ -0,0 +1,66 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerAdvancementDoneEvent;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+
+public class EvtAdvancement extends SkriptEvent {
+
+    static {
+        Skript.registerEvent("Advancement done", EvtAdvancement.class, PlayerAdvancementDoneEvent.class, "[on] advancement (done|finish)")
```

Comment:
I think you can register your event in `SimpleEvents.java`: https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/events/SimpleEvents.java

Reply:
Any event values can be registered in https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java

Reply:
The event should be registered conditionally, only if the corresponding Bukkit event exists (Skript.classExists)